gcloud-java-core/src/main/java/com/google/cloud/FieldSelector.java

```diff
@@ -63,34 +63,51 @@ private static String selector(List<? extends FieldSelector> required, FieldSele
     }
 
     /**
-     * Returns a composite selector given a number of fields. The string selector returned by this
-     * method can be used for field selection in API calls that return a single resource. This
-     * method is not supposed to be used directly by users.
+     * Returns a composite selector given a number of resource fields. The string selector returned
+     * by this method can be used for field selection in API calls that return a single resource.
+     * This method is not supposed to be used directly by users.
      */
     public static String selector(List<? extends FieldSelector> required, FieldSelector... others) {
       return selector(required, others, new String[]{});
     }
 
     /**
-     * Returns a composite selector given a number of fields and a container name. The string
-     * selector returned by this method can be used for field selection in API calls that return a
-     * list of resources. This method is not supposed to be used directly by users.
+     * Returns a composite selector given a number of resource fields and a container name. The
+     * string selector returned by this method can be used for field selection in API calls that
+     * return a  list of resources. This method is not supposed to be used directly by users.
      */
     public static String listSelector(String containerName, List<? extends FieldSelector> required,
         FieldSelector... others) {
       return "nextPageToken," + containerName + '(' + selector(required, others) + ')';
     }
 
     /**
-     * Returns a composite selector given a number of fields and a container name. This methods also
-     * takes an {@code extraResourceFields} parameter to specify some extra fields as strings. The
-     * string selector returned by this method can be used for field selection in API calls that
-     * return a list of resources. This method is not supposed to be used directly by users.
+     * Returns a composite selector given a number of resource fields and a container name. This
+     * method also takes an {@code extraResourceFields} parameter to specify some extra resource
+     * fields as strings. The string selector returned by this method can be used for field
+     * selection in API calls that return a list of resources. This method is not supposed to be
+     * used directly by users.
      */
     public static String listSelector(String containerName, List<? extends FieldSelector> required,
         FieldSelector[] others, String... extraResourceFields) {
       return "nextPageToken," + containerName + '('
```

Comment:
Fine as is. Another option to consider is to delegate:
```java
      return listSelector(new String[0], containerName, required, others, extraResourceFields);
```
or use `null` instead of empty string array and deal with it in the other method.